File: 剑指 Offer（第 2 版）/05.cpp
// 剑指 Offer 05. 替换空格
// https://leetcode-cn.com/problems/ti-huan-kong-ge-lcof/

#include <string>
using namespace std;

class Solution {
public:
    string replaceSpace(string s) {
        string ans;
        // cpp 中 string 的特性
        for(char &c:s){
            if(c == ' '){
                ans += "%20";
            }else{
                ans += c;
            }
        }
        return ans;
    }
};

File: 剑指 Offer（第 2 版）/05.java
class Solution {
    public String replaceSpace(String s) {
        StringBuilder ans = new StringBuilder();
        for (Character c: s.toCharArray()) {
            if (c==' ') ans.append("%20");
            else ans.append(c);
        }
        return ans.toString();
        // 或者直接 return s.replace(" ", "%20");
    }
}